src/misc/ansi_code_markup.cpp

```diff
@@ -0,0 +1,200 @@
+/*
+ *  SPDX-License-Identifier: GPL-2.0-or-later
+ *
+ *  Copyright (C) 2022  The DOSBox Staging Team
+ *
+ *  This program is free software; you can redistribute it and/or modify
+ *  it under the terms of the GNU General Public License as published by
+ *  the Free Software Foundation; either version 2 of the License, or
+ *  (at your option) any later version.
+ *
+ *  This program is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ *  GNU General Public License for more details.
+ *
+ *  You should have received a copy of the GNU General Public License along
+ *  with this program; if not, write to the Free Software Foundation, Inc.,
+ *  51 Franklin Street, Fifth Floor, Boston, MA 02110-1301, USA.
+ */
+
+#include <regex>
+#include <unordered_map>
+
+#include "string_utils.h"
+#include "ansi_code_markup.h"
+
+template <typename K, typename V>
+static bool map_contains(std::unordered_map<K, V> &map, K key)
+{
+	return map.find(key) != map.end();
+}
+
```

Comment:
In support, we have a template-searcher for array-containers called `contains(...)` (it works for lists, vectors, queues, and sets). 

This would make a nice addition for maps, if you'd like to call it `contains(..)` as well - its interface is exactly like the existing one.

I also find the `!= map.end()` check very harsh on the eyes.  But being able to use `if(contains(bucket, item))` is much nicer :-).

Reply:
Yeah, bring on C++20 with it's `contains()` method. Why it wasn't there in the first place is a bit baffling.

Reply:
I wrote an implementation of `contains` in `support.h` that works for containers with `key_type`

Reply:
Such a small change, yet opens up `contains` for (I think?) every STL container now.